### PR TITLE
refactor(presentation): Create a new npm package for core presentation modules in deck

### PR DESCRIPTION
**Pull request by vigneshm**
Kickstarting a new package for presentation modules in deck. This should contain many of the core re-usable UI components (and their dependencies) currently available under `@spinnaker/core`. The goal of this effort is to enable plugin developers build their UI experience with minimal dependence on `@spinnaker/core`.

### CODE REVIEW

File: packages/presentation/src/utils/noop.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+function noopFn() {}
+const noop = noopFn as any;
+export { noop };
```
Comment (caseyhebebrand) [inline packages/presentation/src/utils/noop.ts:3]: Is this just to avoid repeating code in potential future noops?

**Review by caseyhebebrand — APPROVED** [review]

File: packages/presentation/src/utils/noop.ts (hunk 1 above)
Comment (vigneshm) [reply packages/presentation/src/utils/noop.ts:3]: Yep. I copied this from the existing implementation in `@spinnaker/core` as an example.

File: packages/presentation/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  "scripts": {
+    "build": "rollup -c",
+    "clean": "npx shx rm -rf dist",
+    "prepublishOnly": "npm run build",
+    "watch": "rollup -c -w"
+  },
+  "peerDependencies": {
+    "lodash": "^4.17.20",
+    "react": "~16.14.0",
+    "react-dom": "^16.14.0"
```
Comment (christopherthielen) [inline packages/presentation/package.json:17]: Wondering if we really need `react-dom` in this package?
@vigneshm

Comment (vigneshm) [reply packages/presentation/package.json:17]: Good call. I don't think we need it. I'll remove it for now and add it back when needed.

**Pull request merged**